### PR TITLE
Add knative-serving CI on 4.10.

### PR DESCRIPTION
**Pull request by Kaustubh-pande**
Adds knative-serving CI to run on OCP 4.10.

/cc @nak3  @markusthoemmes @mgencur

### CODE REVIEW

**Comment by nak3** [comment]
/lgtm
/hold

Hold until test passed. Thank you @Kaustubh-pande !

File: ci-operator/config/openshift/knative-serving/openshift-knative-serving-release-next__4.10.yaml [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+test_binary_build_commands: make test-install
+tests:
+- as: e2e-aws-ocp-410
+  cluster_claim:
+    architecture: amd64
+    cloud: aws
+    owner: openshift-ci
+    product: ocp
+    timeout: 1h0m0s
+    version: "4.8"
```
Comment (markusthoemmes) [inline ci-operator/config/openshift/knative-serving/openshift-knative-serving-release-next__4.10.yaml:247]: this is wrong I believe. It has to be 4.10. If this is the same in 4.9, we have a bug.

Comment (nak3) [reply ci-operator/config/openshift/knative-serving/openshift-knative-serving-release-next__4.10.yaml]: Ugh... Probably we need to wait for 4.10 is available in https://github.com/openshift/release/tree/master/clusters/hive/pools/openshift-ci/ ?

Comment (markusthoemmes) [reply ci-operator/config/openshift/knative-serving/openshift-knative-serving-release-next__4.10.yaml]: Or alternatively run without pooling for 4.10 while there's no pool yet.

Comment (nak3) [reply ci-operator/config/openshift/knative-serving/openshift-knative-serving-release-next__4.10.yaml]: Okay, let's ping to test-platform team on Slack if they can easily add the pool for 4.10 or not. If not, let's run without pooling for 4.10.

Comment (nak3) [reply ci-operator/config/openshift/knative-serving/openshift-knative-serving-release-next__4.10.yaml]: As per test-platform team, 4.10 is not available until stable 4.10 is available in https://amd64.ocp.releases.ci.openshift.org/#4-stable

@Kaustubh-pande Could you try to run without pooling for 4.10? I think you can refer to other repo's setting or this PR https://github.com/openshift/release/pull/22681

**Comment by mgencur** [comment]
The cluster-pool for 4.10 will be available not before end of Jan. Not sure you want to wait so long. You could change the configuration to avoid cluster pools (similar to what we have on OCP 4.7 an older)

**Comment by nak3** [comment]
@Kaustubh-pande 4.10 pool should be available now https://github.com/openshift/release/blob/master/clusters/hive/pools/openshift-ci/ci-ocp-4-10-0-amd64-aws-us-east-1_clusterpool.yaml 
Could you please update this PR?

**Pull request merged**